src/valkey-benchmark.c

```diff
@@ -1160,6 +1163,55 @@ static void showLatencyReport(void) {
         printf("  latency summary (msec):\n");
         printf("    %9s %9s %9s %9s %9s %9s\n", "avg", "min", "p50", "p95", "p99", "max");
         printf("    %9.3f %9.3f %9.3f %9.3f %9.3f %9.3f\n", avg, p0, p50, p95, p99, p100);
+
+        /* RPS analysis - always show when --rps is used */
+        if (config.rps > 0) {
+            float target_rps = (float)config.rps;
+            float achievement_pct = (reqpersec / target_rps) * 100.0f;
+
+            printf("\n  RPS Analysis:\n");
+            printf("    Target RPS: %.0f\n", target_rps);
+            printf("    Actual RPS: %.2f (%.1f%% of target)\n", reqpersec, achievement_pct);
+
+            // ADD RPS PERCENTILE ANALYSIS
+            if (config.rps_histogram && config.rps_histogram->total_count > 0) {
+                const float rps_p1 = hdr_value_at_percentile(config.rps_histogram, 1.0);
+                const float rps_p10 = hdr_value_at_percentile(config.rps_histogram, 10.0);
+                const float rps_p50 = hdr_value_at_percentile(config.rps_histogram, 50.0);
+                // const float rps_p90 = hdr_value_at_percentile(config.rps_histogram, 90.0);
+                // const float rps_p99 = hdr_value_at_percentile(config.rps_histogram, 99.0);
+
+                printf("    RPS Performance Guarantees:\n");
+                printf("      99%% of time: RPS ≥ %.0f (%.1f%% of target)\n",
+                       rps_p1, (rps_p1 / target_rps) * 100);
+                printf("      90%% of time: RPS ≥ %.0f (%.1f%% of target)\n",
+                       rps_p10, (rps_p10 / target_rps) * 100);
+                printf("      50%% of time: RPS ≥ %.0f (%.1f%% of target)\n",
+                       rps_p50, (rps_p50 / target_rps) * 100);
+
+                // Bottleneck analysis using P1 (99% of time achieved this or better)
+                if (rps_p1 >= 0.99 * target_rps) {
+                    printf("    Status: GOOD PERFORMANCE\n");
```

Comment:
I'm not sure if no bottleneck always means good performance. If we tested with a very low RPS, we don't know much.